tests/lib/ocpp/v201/test_charge_point.cpp

```diff
@@ -711,4 +711,49 @@ TEST_F(ChargePointFixture, K01FR22_SetChargingProfileRequest_RejectsChargingStat
     charge_point->handle_message(set_charging_profile_req);
 }
 
+/// This test ensures that when Smart Charging isn't available it doesn't process the submitted profile.
+///
+/// This test is disabled because it only passes if the ControllerComponentVariables::SmartChargingCtrlrAvailable
+/// entry is removed from the config/v201/config.json file.
+TEST_F(ChargePointFixture, DISABLED_K01FR29_SmartChargingCtrlrAvailableIsFalse_RespondsCallError) {
```

Comment:
Why can't you use `this->device_model->set_value` to set `SmartChargingCtrlr.Enabled` to false for this test?

Reply:
Take a look at where `set_value` is used in `lib/ocpp/v201/charge_point.cpp`. I would expect something like the following to work here:

```c++
this->device_model->set_value(
        ControllerComponentVariables::SmartChargingCtrlrAvailableEnabled.component,
        ControllerComponentVariables::SmartChargingCtrlrAvailableEnabled.variable.value(),
        AttributeEnum::Actual,
        "false",  // New variable value
        "internal",  // Source of variable value
        true  // Allow write even if variable is read-only
)
```

Reply:
Spent some time try to get this to work with no luck. Want to focus on this later.